components/imap_ops.py

```
@@ -0,0 +1,314 @@
+                if "@" in account.email:
+                    local, domain = account.email.split("@", 1)
+                else:
+                    raise ValueError("invalid email address for DA provisioning")
+                # Create mailbox then retry login
+                client.create_pop_account(domain, local, account.password, quota_mb=quota_mb)  # type: ignore[attr-defined]
+                logging.info("[da][lazy] Created mailbox: %s, retrying login", account.email)
+                _try_login_only()
+                login_ok = True
+            except Exception:
+                # Propagate original login failure if provisioning/retry fails
+                raise first_exc
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)